plugins/tech/azure/src/main/java/org/apache/hop/vfs/azure/AzureFileObject.java

```diff
@@ -88,138 +87,156 @@ public void close() throws IOException {
-            for (ListBlobItem item :
-                relpath.equals("") ? container.listBlobs() : container.listBlobs(relpath + "/")) {
-              String itemPath = removeTrailingSlash(item.getUri().getPath());
-              if (pathsMatch(itemPath, thisPath)) {
-                if (item instanceof CloudBlob) {
-                  cloudBlob = (CloudBlob) item;
-                } else {
-                  cloudDir = (CloudBlobDirectory) item;
-                  for (ListBlobItem blob : cloudDir.listBlobs()) {
-                    URI blobUri = blob.getUri();
-                    String path = blobUri.getPath();
-                    while (path.endsWith("/")) path = path.substring(0, path.length() - 1);
-                    int idx = path.lastIndexOf('/');
-                    if (idx != -1) path = path.substring(idx + 1);
-                    children.add(path);
-                  }
-                }
-                break;
-              }
-            }
-            if (cloudBlob != null) {
-              type = FileType.FILE;
-              size = cloudBlob.getProperties().getLength();
-              if (cloudBlob.getMetadata().containsKey("ActualLength")) {
-                size = Long.parseLong(cloudBlob.getMetadata().get("ActualLength"));
-              }
-              String disp = cloudBlob.getProperties().getContentDisposition();
-              if (disp != null && disp.startsWith("vfs ; length=\"")) {
-                size = Long.parseLong(disp.substring(14, disp.length() - 1));
-              }
-              Date lastModified2 = cloudBlob.getProperties().getLastModified();
-              lastModified = lastModified2 == null ? 0 : lastModified2.getTime();
-            } else if (cloudDir != null) {
-              type = FileType.FOLDER;
-              size = children.size();
-              lastModified = 0;
-            } else {
-              lastModified = 0;
-              type = FileType.IMAGINARY;
-              size = 0;
-            }
+            lastModified = 0;
+            type = FileType.IMAGINARY;
+            size = 0;
+            pathItem = null;
+            dirPathItem = null;
           }
-        } else {
-          lastModified = 0;
-          type = FileType.IMAGINARY;
-          size = 0;
-          cloudBlob = null;
-          cloudDir = null;
         }
       }
     }
   }
 
-  private boolean pathsMatch(String path, String currentPath) {
-    return path.replace("/" + ((AzureFileSystem) getFileSystem()).getAccount(), "")
-        .equals(currentPath);
+  private boolean containerExists() {
+    String containerName = ((AzureFileName) getName()).getContainer();
+    ListFileSystemsOptions fileSystemsOptions = new ListFileSystemsOptions();
+    fileSystemsOptions.setPrefix(containerName);
+
+    final DataLakeFileSystemClient fileSystemClient = service.getFileSystemClient(containerName);
+
+    try {
+      return fileSystemClient.existsWithResponse(Duration.ofSeconds(5), Context.NONE).getValue();
```

Comment:
we need to check whether the container/filesystem already exist, in order to go ahead or throw an exception;  since `fileSystemClient.exist()` returns in a few seconds when the container exists, but the request is left hanging without an answer when the container does not exist, I've adopted this workaround. 
This fact doesn't happen in our sandbox, so it must depend on the dependencies we bring in Hop. 
Of course it's awful, but I couldn't see any alternative without resolving all the dependency issues. When we're all set, we can get rid of the timeout logic